### PR TITLE
Update to use ethers v6 and update chains config from dist.

### PR DESCRIPTION
**Pull request by iainnash**

### CODE REVIEW

**Comment by iainnash** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #16** <a href="https://app.graphite.dev/github/pr/ourzora/smol-safe/16?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ourzora/smol-safe/16?utm_source=stack-comment).

File: package.json [hunk 1]
```diff
@@ -4,13 +4,10 @@
   "license": "MIT",
   "type": "module",
   "dependencies": {
-    "@ethersproject/contracts": "^5.7.0",
-    "@ethersproject/providers": "^5.7.2",
-    "@ethersproject/wallet": "^5.7.0",
-    "@safe-global/protocol-kit": "^1.0.1",
-    "@safe-global/safe-deployments": "^1.25.0",
+    "@safe-global/protocol-kit": "^2.0.0",
```
Comment (oveddan) [inline package.json:7]: oh new sdk? nice!

File: src/chains.ts [hunk 2]
```diff
@@ -69,9 +60,14 @@ const pgn = {
...
-  [999]: chains.zoraTestnet,
-  [999999999]: chains.zoraSepolia,
-  [424]: pgn,
+  [chains.zoraTestnet.id]: chains.zoraTestnet,
+  [chains.zoraSepolia.id]: chains.zoraSepolia,
+  [chains.arbitrumGoerli.id]: chains.arbitrumGoerli,
+  [chains.arbitrumSepolia.id]: chains.arbitrumSepolia,
+  [chains.arbitrum.id]: chains.arbitrum,
+  [chains.base.id]: chains.base,
+  [chains.baseSepolia.id]: chains.baseSepolia,
```
Comment (oveddan) [inline src/chains.ts:69]: could you add op chains too while were at it?

**Review by oveddan — APPROVED** [review]
just add op chains then LGTM

File: src/chains.ts [hunk 3]
```diff
@@ -19,28 +21,17 @@ const baseL2Addresses = {
...
   createCallAddress: "0xB19D6FFc2182150F8Eb585b79D4ABcd7C5640A9d",
   signMessageLibAddress: "0x98FFBBF51bb33A056B08ddf711f289936AafF717",
+  safeSingletonAddress: "0x",
+  simulateTxAccessorAddress: "0x59AD6735bCd8152B84860Cb256dD9e96b85F69Da",
 };
 
-// sepolia is the same as base.
-const sepoliaAddresses = baseL2Addresses;
-
 export const contractNetworks: ContractNetworksConfig = {
```
Comment (oveddan) [inline src/chains.ts:28]: do we need this? can it be commented out?

**Review by kulkarohan — APPROVED** [review]

File: src/chains.ts (hunk 2 above)
Comment (iainnash) [reply src/chains.ts:73]: Sure

File: src/chains.ts [hunk 4]
```diff
@@ -19,68 +21,69 @@ const baseL2Addresses = {
...
-export const allowedNetworks: { [chainId: number]: chains.Chain }= {
-  [999]: chains.zoraTestnet,
-  [999999999]: chains.zoraSepolia,
-  [424]: pgn,
+export const allowedNetworks: { [chainId: number]: chains.Chain } = {
+  [chains.zoraTestnet.id]: chains.zoraTestnet,
+  [chains.zoraSepolia.id]: chains.zoraSepolia,
+  [chains.arbitrumGoerli.id]: chains.arbitrumGoerli,
+  [chains.arbitrumSepolia.id]: chains.arbitrumSepolia,
+  [chains.arbitrumNova.id]: chains.arbitrumNova,
```
Comment (oveddan) [inline src/chains.ts:70]: zora mainnet is missing - can you please add it back in?

**Review by oveddan — CHANGES_REQUESTED** [review]
please add back zora mainnet

File: src/chains.ts (hunk 3 above)
Comment (iainnash) [reply src/chains.ts:28]: It'll be helpful to keep at least one example if we need it.

File: src/chains.ts (hunk 4 above)
Comment (iainnash) [reply src/chains.ts:70]: Resolved in stack.

**Comment by iainnash** [comment]
### Merge activity

* **Jan 30, 12:50 PM**: @iainnash merged this pull request with [Graphite](https://app.graphite.dev/github/pr/ourzora/smol-safe/16).

**Review by oveddan — APPROVED** [review]

**Pull request merged**